sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala

```diff
@@ -47,6 +48,10 @@ abstract class StreamingJoinSuite
     StateStore.stop()
   }
 
+  override protected def sparkConf = super.sparkConf
+    // TODO(SPARK-35095): Use ANSI intervals in streaming join tests
```

Comment:
For some reasons, `CalendarInterval` is used in the streaming code. For instance at
https://github.com/apache/spark/blob/0945baf90660a101ae0f86a39d4c91ca74ae5ee3/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/EventTimeWatermarkExec.scala#L89-L91

Need to revise the code. The streaming tests fails on new ANSI intervals at the moment.